virtcontainers/container.go

```diff
@@ -1206,6 +1219,18 @@ func (c *Container) removeResources() error {
 		}
 	}
 
+	mem := c.config.Resources.Mem
+	if mem != 0 {
+		virtLog.Debugf("hot removing %d B memory", mem)
+		sizeMB := int(mem / 1024 / 1024)
+		// sizeMB needs to be divisible by 2
+		sizeMB = sizeMB + sizeMB%2
+		_, err := c.sandbox.hypervisor.hotplugRemoveDevice(&memoryDevice{1, sizeMB}, memoryDev)
```

Comment:
To support hot-unplug memory devices, we need to support guest ballooning first (which I don't think we have added yet).